tests/test_exceptions.py

```diff
@@ -27,10 +27,10 @@ def test_http_request_error_init():
 
     response = HttpResponse(URL, b"")
     with pytest.raises(TypeError):
-        HttpRequestError(request=request, response=response)
+        HttpRequestError(request=request, response=response)  # type: ignore[call-arg]
```

Comment:
This seems like a valid warning from mypy.

It seems HttpRequestError expects (msg, request) as parameters. Maybe we need to define `__init__` in HttpRequestError to allow an optional `response` parameter?

https://github.com/scrapinghub/web-poet/blob/master/web_poet/exceptions/http.py#L12-L40

Reply:
Isn't this code checking a TypeError because of incorrect parameters, and typing fails because of that (as it should)?

Reply:
Oh, right! :facepalm: